zetaclient/zetacore/client.go

```diff
@@ -367,87 +362,83 @@ func (c *Client) UpdateAppContext(
 		)
 
 		c.Stop()
-	}
 
-	additionalChains, err := c.GetAdditionalChains(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to additional chains: %w", err)
+		return nil
 	}
 
-	chainParams, err := c.GetChainParams(ctx)
+	supportedChains, err := c.GetSupportedChains(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to get chain params: %w", err)
+		return errors.Wrap(err, "unable to fetch supported chains")
 	}
 
-	newEVMParams := make(map[int64]*observertypes.ChainParams)
-	var newBTCParams *observertypes.ChainParams
-	var newSolanaParams *observertypes.ChainParams
-
-	// check and update chain params for each chain
-	for _, chainParam := range chainParams {
-		err := observertypes.ValidateChainParams(chainParam)
-		if err != nil {
-			sampledLogger.Warn().Err(err).Msgf("Invalid chain params for chain %d", chainParam.ChainId)
-			continue
-		}
-		if chains.IsBitcoinChain(chainParam.ChainId, additionalChains) {
-			newBTCParams = chainParam
-		} else if chains.IsSolanaChain(chainParam.ChainId, additionalChains) {
-			newSolanaParams = chainParam
-		} else if chains.IsEVMChain(chainParam.ChainId, additionalChains) {
-			newEVMParams[chainParam.ChainId] = chainParam
-		}
+	additionalChains, err := c.GetAdditionalChains(ctx)
+	if err != nil {
+		return errors.Wrap(err, "unable to fetch additional chains")
 	}
 
-	supportedChains, err := c.GetSupportedChains(ctx)
+	chainParams, err := c.GetChainParams(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to get supported chains: %w", err)
+		return errors.Wrap(err, "unable to fetch chain params")
 	}
 
-	newChains := make([]chains.Chain, len(supportedChains))
-	for i, chain := range supportedChains {
-		newChains[i] = chain
+	keyGen, err := c.GetKeyGen(ctx)
+	if err != nil {
+		return errors.Wrap(err, "unable to fetch keygen from zetacore")
 	}
 
-	keyGen, err := c.GetKeyGen(ctx)
+	crosschainFlags, err := c.GetCrosschainFlags(ctx)
 	if err != nil {
-		c.logger.Info().Msg("Unable to fetch keygen from zetacore")
-		return fmt.Errorf("failed to get keygen: %w", err)
+		return errors.Wrap(err, "unable to fetch crosschain flags from zetacore")
 	}
 
 	tss, err := c.GetCurrentTSS(ctx)
 	if err != nil {
-		c.logger.Info().Err(err).Msg("Unable to fetch TSS from zetacore")
-		return fmt.Errorf("failed to get current tss: %w", err)
+		return errors.Wrap(err, "unable to fetch current TSS")
 	}
-	tssPubKey := tss.GetTssPubkey()
 
-	crosschainFlags, err := c.GetCrosschainFlags(ctx)
-	if err != nil {
-		c.logger.Info().Msg("Unable to fetch cross-chain flags from zetacore")
-		return fmt.Errorf("failed to get crosschain flags: %w", err)
+	var (
+		freshChains = make([]chains.Chain, 0, len(supportedChains))
+		freshParams = make(map[int64]*observertypes.ChainParams, len(chainParams))
+	)
+
+	// check and update chain params for each chain
+	for chainID := range chainParams {
```

Comment:
is `chainID` accurate ?

Reply:
What do you mean in particular?

Reply:
we are not ranging over chainIDs here; it is the index of the chainParams array; chainid is a value in chain params?

Reply:
ignore that , I see you already fixed it